pkg/agent/agent_windows.go

```diff
@@ -111,6 +117,104 @@ func (i *Initializer) prepareHNSNetworkAndOVSExtension() error {
 	return util.PrepareHNSNetwork(subnetCIDR, i.nodeConfig.NodeTransportIPv4Addr, adapter, i.nodeConfig.UplinkNetConfig.Gateway, dnsServers, i.nodeConfig.UplinkNetConfig.Routes, i.ovsBridge)
 }
 
+func (i *Initializer) prepareVmNetworkAndOVSExtension() error {
+	var eNodeIface *net.Interface
+	en, err := i.crdClient.CrdV1alpha1().ExternalNodes(i.namespace).Get(context.TODO(), i.GetNodeConfig().Name, metav1.GetOptions{})
+	if err != nil {
+		klog.InfoS("Failed to get ExternalNode", "Name", i.GetNodeConfig().Name)
+		return err
+	}
+	for _, intF := range en.Spec.Interfaces {
+		//TODO: Handle multiple interface case
+		for _, addr := range intF.IPs {
+			var ipFilter *ip.DualStackIPs
+			epIP := net.ParseIP(addr)
+			if epIP.To4() != nil {
+				ipFilter = &ip.DualStackIPs{IPv4: epIP}
+			} else {
+				ipFilter = &ip.DualStackIPs{IPv6: epIP}
+			}
+			_, _, eNodeIface, err = util.GetIPNetDeviceFromIP(ipFilter, sets.NewString())
+			if err != nil {
+				klog.InfoS("Failed to get net device", "IP", addr)
+			} else {
+				klog.V(2).InfoS("Using the ExternalNode", "InterfaceName", eNodeIface.Name)
+				break
+			}
+		}
+		break
+	}
+
+	klog.V(2).InfoS("Setting up VM network")
+	// Check whether VM Switch is created
+	exists, err := util.VMSwitchExists()
+	if err != nil {
+		return err
+	}
+	if exists {
+		vmSwitchIfName, err := util.GetVMSwitchInterfaceName()
+		if err != nil {
+			return err
+		} else {
+			klog.InfoS("VM switch teaming", "interfaceName", vmSwitchIfName)
+			vmSwitchIfName = strings.TrimPrefix(vmSwitchIfName, util.PrefixPhy)
```

Comment:
Provide a general function in path pkg/agent/util/net to get the interface name, then we don't need to know the name format details in initializer?

Reply:
Sure, will add helper function to get the host interface name from the uplink interface name